### PR TITLE
[mempool] add support to limit maximum bytes for a batch

### PR DESCRIPTION
**Pull request by zekun000**
There's a bump for single transaction size limit to 6MB https://github.com/aptos-labs/aptos-core/pull/2829, there's no way we can support 6mb  * 6000 blocks, this commit adds the limit on bytes level

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2844)

### CODE REVIEW

File: config/src/config/consensus_config.rs [hunk 1]
```diff
@@ -8,7 +8,8 @@ use std::path::PathBuf;
 #[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
 #[serde(default, deny_unknown_fields)]
 pub struct ConsensusConfig {
-    pub max_block_size: u64,
+    pub max_block_txn_size: u64,
```
Comment (msmouse) [inline config/src/config/consensus_config.rs:11]: "max_block_txns", "max_block_bytes"?

File: consensus/src/payload_manager.rs [hunk 2]
```diff
@@ -79,7 +84,8 @@ impl QuorumStoreClient {
 impl PayloadManager for QuorumStoreClient {
     async fn pull_payload(
         &self,
-        max_size: u64,
+        max_item_size: u64,
```
Comment (msmouse) [inline consensus/src/payload_manager.rs:87]: max_items, max_bytes

File: mempool/src/tests/core_mempool_test.rs [hunk 3]
```diff
@@ -672,3 +696,20 @@ fn test_get_transaction_by_hash_after_the_txn_is_updated() {
...
+    // add 100 transacionts
+    for seq in 0..100 {
+        add_txn(&mut pool, TestTransaction::new(1, seq, 1)).unwrap();
+    }
+    let get_all = pool.get_batch(100, 100 * 1024, HashSet::new());
+    assert_eq!(get_all.len(), 100);
+    let txn_size = get_all[0].raw_txn_bytes_len() as u64;
+    let limit = 10;
+    let hit_limit = pool.get_batch(100, txn_size * limit, HashSet::new());
+    assert_eq!(hit_limit.len(), limit as usize);
```
Comment (msmouse) [inline mempool/src/tests/core_mempool_test.rs:714]: not really testing the bytes limit being hit?

**Comment by msmouse** [comment]
Looks reasonable to me..

File: mempool/src/tests/core_mempool_test.rs (hunk 3 above)
Comment (zekun000) [reply mempool/src/tests/core_mempool_test.rs:714]: it only returns 10 txns instead of 100?

Comment (msmouse) [reply mempool/src/tests/core_mempool_test.rs:714]: ugh. for sure. I'm stupid.

File: consensus/src/liveness/proposal_generator.rs [hunk 4]
```diff
@@ -46,11 +45,13 @@ pub struct ProposalGenerator {
     // Time service to generate block timestamps
     time_service: Arc<dyn TimeService>,
     // Max number of transactions to be added to a proposed block.
-    max_block_size: u64,
+    max_block_txn_size: u64,
```
Comment (msmouse) [inline consensus/src/liveness/proposal_generator.rs:48]: max_block_txns.. how about search and replace all?

**Review by msmouse — APPROVED** [review]

File: config/src/config/consensus_config.rs [hunk 5]
```diff
@@ -32,7 +33,8 @@ pub struct ConsensusConfig {
 impl Default for ConsensusConfig {
     fn default() -> ConsensusConfig {
         ConsensusConfig {
-            max_block_size: 6000,
+            max_block_txns: 6000,
+            max_block_bytes: 5 * 1024 * 1024, // 5MB
```
Comment (sitalkedia) [inline config/src/config/consensus_config.rs:37]: This number looks small to me. How did you get this number? Do you have a rough estimate of how many transactions on average we can pack with this? Ideally this should allow 4-5k average-sized txns?

**Review by bchocho — COMMENTED** [review]
Some typos

**Pull request merged**